jbmc/unit/java_bytecode/java_bytecode_parser/parse_inner_class.cpp

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting inner classes
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <java-testing-utils/load_java_class.h>
+#include <java_bytecode/java_types.h>
+#include <testing-utils/catch.hpp>
+
+SCENARIO(
+  "Parsing inner java classes",
+  "[core][java_bytecode][java_bytecode_parser]")
+{
+  GIVEN("A class file of an inner class.")
+  {
+    WHEN("Parsing an inner class called \"Inner\".")
+    {
+      const symbol_tablet &symbol_table = load_java_class(
+        "Outer$Inner", "./java_bytecode/java_bytecode_parser");
+      const symbolt *class_symbol = symbol_table.lookup("java::Outer$Inner");
+      REQUIRE(class_symbol);
+      const java_class_typet *class_type =
+        type_try_dynamic_cast<java_class_typet>(class_symbol->type);
+      REQUIRE(class_type);
+
+      THEN("The inner class should have the inner name \"Inner\".")
+      {
+        REQUIRE(id2string(class_type->get_inner_name()) == "Inner");
```

Comment:
:pick: Add a check on `name` as it isn't obvious to me why this isn't the name of the class